testframework/src/main/java/net/neoforged/testframework/summary/GitHubActionsStepSummaryFormatter.java

```diff
@@ -0,0 +1,94 @@
+        this("Test Summary");
+    }
+
+    public GitHubActionsStepSummaryFormatter(String heading) {
+        this(summary -> heading);
+    }
+
+    public GitHubActionsStepSummaryFormatter(Function<TestSummary, String> heading) {
+        this.heading = heading;
+    }
+
+    @Override
+    public Path outputPath(ResourceLocation frameworkId) {
+        return Path.of(System.getenv("GITHUB_STEP_SUMMARY"));
+    }
+
+    @Override
+    public boolean enabled(TestSummary summary) {
+        return summary.isGameTestRun() && System.getenv().containsKey("GITHUB_STEP_SUMMARY");
+    }
+
+    @Override
+    public void write(TestSummary summary, Logger logger, PrintWriter writer) {
+        writer.println("# " + this.heading.apply(summary));
+        Map<Test.Result, List<TestSummary.TestInfo>> testsByStatus = summary.testInfos()
+                .stream()
+                .collect(Collectors.groupingBy(test -> test.status().result(), () -> new EnumMap<>(Test.Result.class), Collectors.toList()));
+        List<TestSummary.TestInfo> failedTests = testsByStatus.getOrDefault(Test.Result.FAILED, List.of());
+        List<TestSummary.TestInfo> passedTests = testsByStatus.getOrDefault(Test.Result.PASSED, List.of());
+        Table.Builder builder = Table.builder()
+                .withAlignments(Alignment.LEFT, Alignment.CENTER, Alignment.LEFT, Alignment.LEFT)
+                .addRow("Test Id", "Test Result", "Status message", "Test description");
+        if (!failedTests.isEmpty()) {
+            for (TestSummary.TestInfo failedTest : failedTests) {
+                builder.addRow(
+                        failedTest.testId(),
+                        formatStatus(failedTest.result(), !failedTest.manual() && !failedTest.required()),
+                        failedTest.status().message(),
+                        getDescription(failedTest));
+            }
+        }
+        if (!passedTests.isEmpty()) {
+            for (TestSummary.TestInfo passedTest : passedTests) {
+                builder.addRow(
+                        passedTest.testId(),
+                        formatStatus(passedTest.status().result(), false),
+                        passedTest.status().message(),
+                        getDescription(passedTest));
+            }
+        }
+        if (!passedTests.isEmpty() && failedTests.isEmpty()) {
+            writer.println("All tests passed");
+        }
+        writer.println();
+        writer.println(builder.build());
+    }
+
+    protected String formatStatus(Test.Result result, boolean optional) {
+        if (result.failed() && !optional) {
+            return "❌";
+        } else if (result.passed()) {
+            return "✅";
+        }
+        return "⚠️";
+    }
+
+    private static String getDescription(TestSummary.TestInfo failedTest) {
+        return FormattingUtil.componentsToPlainString(failedTest.description().stream().filter(c -> !c.getString().equals("GameTest-only")).toList());
```

Comment:
This looks so hacky, it's much better to add a `boolean exclusive` to `GameTestData` and default it to `true` when the test is gametest-based.

Reply:
what do you mean I am filtering out the "GameTest-only" line that is added to the description by default

Reply:
Ah wait, I didn't actually realise that, thought you were trying to see if a test is gametest exclusive. whoops